# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/android_video_capture/video_capture_android.h
# Commit: 9a2171028b31
# Full Hash: 9a2171028b314fd9f9b3d06aced40cfd6b28aec4
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2020-02-13 18:21:33
# Regressor Bug: 1578073
# File Overlap Count: 2
# Description:
#   Bug 1578073 - Use updated camera capture code; r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D61861
# ==============================================================================

diff -r 7a9b07dec9f9 -r 9a2171028b31 dom/media/systemservices/android_video_capture/video_capture_android.h
--- a/dom/media/systemservices/android_video_capture/video_capture_android.h	Tue Feb 11 17:07:22 2020 +0000
+++ b/dom/media/systemservices/android_video_capture/video_capture_android.h	Tue Feb 11 17:08:15 2020 +0000
@@ -14,6 +14,7 @@
 #include <jni.h>
 
 #include "device_info_android.h"
+#include "api/video/i420_buffer.h"
 #include "modules/video_capture/video_capture_impl.h"
 
 namespace webrtc {
@@ -28,10 +29,9 @@
   virtual int32_t StopCapture();
   virtual bool CaptureStarted();
   virtual int32_t CaptureSettings(VideoCaptureCapability& settings);
-  virtual int32_t SetCaptureRotation(VideoRotation rotation);
 
-  int32_t OnIncomingFrame(uint8_t* videoFrame, size_t videoFrameLength,
-                          int32_t degrees, int64_t captureTime = 0);
+  void OnIncomingFrame(rtc::scoped_refptr<I420Buffer> buffer, int32_t degrees,
+                       int64_t captureTime = 0);
 
  protected:
   virtual ~VideoCaptureAndroid();
@@ -39,7 +39,6 @@
   DeviceInfoAndroid _deviceInfo;
   jobject _jCapturer;  // Global ref to Java VideoCaptureAndroid object.
   VideoCaptureCapability _captureCapability;
-  VideoRotation _rotation;
   bool _captureStarted;
 };
 
